FnCAS/fncas/differentiate.h

```diff
@@ -33,32 +33,32 @@
 
 namespace fncas {
 
-static const double APPROXIMATE_DERIVATIVE_EPS = 1e-4;
+static const double_t APPROXIMATE_DERIVATIVE_EPS = 1e-4;
 
 template <typename F>
-fncas_value_type approximate_derivative(F f,
-                                        const std::vector<fncas_value_type>& x,
-                                        node_index_type i,
-                                        const fncas_value_type EPS = APPROXIMATE_DERIVATIVE_EPS) {
-  std::vector<fncas_value_type> x1(x);
-  std::vector<fncas_value_type> x2(x);
+double_t approximate_derivative(F f,
```

Comment:
It's not true, however not relevant in this particular case. 
I've suggested it for consistency reasons, especially that there's `inline std::vector<double_t> approximate_gradient` below.